packages/lib/services/rest/routes/notes.ts

```diff
@@ -262,27 +264,32 @@ export async function downloadMediaFile(url: string, fetchOptions?: FetchOptions
 	}
 }
 
-async function downloadMediaFiles(urls: string[], fetchOptions?: FetchOptions, allowedProtocols?: string[]) {
-	const PromisePool = require('es6-promise-pool');
-
+async function downloadMediaFiles(urls: string[], fetchOptions: FetchOptions, allowedProtocols?: string[]) {
 	const output: any = {};
 
+	const downloadController = fetchOptions?.downloadController ?? null;
```

Comment:
fetchOptions is required but there's a check if it's defined here? Or shouldn't fetchOptions be optional? (it previously was and I'm not sure why you changed it)

Reply:
It should be optional, thanks for catching this!